### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by elharo**
Remove unused variables and deprecated methods
@khmarbaise

### CODE REVIEW

File: src/test/java/org/apache/maven/plugins/dependency/resolvers/ExcludeReactorProjectsDependencyFilterTest.java [hunk 1]
```diff
@@ -79,9 +78,7 @@ public Dependency getDependency() {
             }
         };
 
-        final boolean result = filter.accept( node , Collections.<Node>emptyList() );
-
-        assertThat( result, is( false ));
+        assertFalse( filter.accept( node , Collections.<Node>emptyList() ) );
```
Comment (Tibor17) [inline src/test/java/org/apache/maven/plugins/dependency/resolvers/ExcludeReactorProjectsDependencyFilterTest.java]: a bad practice, here again.

File: src/test/java/org/apache/maven/plugins/dependency/resolvers/ExcludeReactorProjectsDependencyFilterTest.java [hunk 2]
```diff
@@ -124,7 +121,7 @@ public Dependency getDependency() {
 
         ArgumentCaptor<String> captor = ArgumentCaptor.forClass( String.class );
         verify( log ).debug( captor.capture() );
-        assertThat( captor.getValue(), containsString( "Skipped dependency" ) );
+        MatcherAssert.assertThat( captor.getValue(), containsString( "Skipped dependency" ) );
```
Comment (Tibor17) [inline src/test/java/org/apache/maven/plugins/dependency/resolvers/ExcludeReactorProjectsDependencyFilterTest.java]: statis import

File: src/test/java/org/apache/maven/plugins/dependency/resolvers/ExcludeReactorProjectsDependencyFilterTest.java (hunk 1 above)
Comment (elharo) [reply src/test/java/org/apache/maven/plugins/dependency/resolvers/ExcludeReactorProjectsDependencyFilterTest.java]: Can you be more specific?

File: src/test/java/org/apache/maven/plugins/dependency/resolvers/ExcludeReactorProjectsDependencyFilterTest.java (hunk 2 above)
Comment (elharo) [reply src/test/java/org/apache/maven/plugins/dependency/resolvers/ExcludeReactorProjectsDependencyFilterTest.java]: done

**Comment by jira-importer** [comment]
Resolve #1254

**Pull request merged**